code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -770,19 +790,636 @@ nobiliumsuppression = INFINITY
+	var/energy_used = heat_efficency * 100
+	ASSERT_GAS(/datum/gas/proto_nitrate, air)
+	if ((cached_gases[/datum/gas/pluoxium][MOLES] - heat_efficency * 0.2 < 0 ) || (cached_gases[/datum/gas/hydrogen][MOLES] - heat_efficency * 2 < 0)) //Shouldn't produce gas from nothing.
+		return NO_REACTION
+	if(air.temperature >= 10000)
+		return NO_REACTION
+	cached_gases[/datum/gas/hydrogen][MOLES] -= heat_efficency * 2
+	cached_gases[/datum/gas/pluoxium][MOLES] -= heat_efficency * 0.2
+	cached_gases[/datum/gas/proto_nitrate][MOLES] += heat_efficency * 2.2
+
+	if(energy_used > 0)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature * old_heat_capacity + energy_used) / new_heat_capacity), TCMB)
+	return REACTING
+
+/datum/gas_reaction/cyrion_b_formation
+	priority = 14
+	name = "Cyrion B formation"
+	id = "cyrion_b_formation"
+
+/datum/gas_reaction/cyrion_b_formation/init_reqs()
+	min_requirements = list(
+		/datum/gas/hypernoblium = MINIMUM_MOLE_COUNT,
+		/datum/gas/stimulum = MINIMUM_MOLE_COUNT,
+		"TEMP" = 260,
+		"MAX_TEMP" = 290
+	)
+
+/datum/gas_reaction/cyrion_b_formation/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_efficency = min(temperature * 0.005, cached_gases[/datum/gas/hypernoblium][MOLES], cached_gases[/datum/gas/stimulum][MOLES])
+	var/energy_used = heat_efficency * 10
+	ASSERT_GAS(/datum/gas/cyrion_b, air)
+	if ((cached_gases[/datum/gas/hypernoblium][MOLES] - heat_efficency * 0.01 < 0 ) || (cached_gases[/datum/gas/stimulum][MOLES] - heat_efficency * 0.5 < 0)) //Shouldn't produce gas from nothing.
+		return NO_REACTION
+	if(air.temperature >= 290)
+		return NO_REACTION
+	cached_gases[/datum/gas/hypernoblium][MOLES] -= heat_efficency * 0.01
+	cached_gases[/datum/gas/stimulum][MOLES] -= heat_efficency * 0.5
+	cached_gases[/datum/gas/cyrion_b][MOLES] += heat_efficency * 0.5
+
+	if(energy_used)
+		var/new_heat_capacity = air.heat_capacity()
+		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
+			air.temperature = max(((temperature * old_heat_capacity + energy_used) / new_heat_capacity), TCMB)
+	return REACTING
+
+/datum/gas_reaction/halon_o2removal
+	priority = -1
+	name = "Halon o2 removal"
+	id = "halon_o2removal"
+
+/datum/gas_reaction/halon_o2removal/init_reqs()
+	min_requirements = list(
+		/datum/gas/halon = MINIMUM_MOLE_COUNT,
+		/datum/gas/oxygen = MINIMUM_MOLE_COUNT,
+		"TEMP" = FIRE_MINIMUM_TEMPERATURE_TO_EXIST
+	)
+
+/datum/gas_reaction/halon_o2removal/react(datum/gas_mixture/air, datum/holder)
+	var/list/cached_gases = air.gases
+	var/temperature = air.temperature
+	var/old_heat_capacity = air.heat_capacity()
+	var/heat_efficency = min(temperature / ( FIRE_MINIMUM_TEMPERATURE_TO_EXIST * 10), cached_gases[/datum/gas/halon][MOLES], cached_gases[/datum/gas/oxygen][MOLES])
```

Comment:
So the hotter it is the better it preforms?

Reply:
yes, the idea is to stop very hot fires fast and easy